crates/bevy_ecs/src/bundle.rs

```diff
@@ -360,11 +360,11 @@ impl BundleInfo {
     /// `entity`, `bundle` must match this [`BundleInfo`]'s type
     #[inline]
     #[allow(clippy::too_many_arguments)]
-    unsafe fn write_components<T: Bundle>(
+    unsafe fn write_components<T: Bundle, S: BundleComponentStatus>(
         &self,
         table: &mut Table,
         sparse_sets: &mut SparseSets,
-        add_bundle: &AddBundle,
+        bundle_component_status: &S,
```

Comment:
The relationship between `T` and `S` is hard to follow here. I'm unsure about whether this should be documented on this method, BundleComponentStatus, or both.

Reply:
I've added clarifying docs on the requirements and behavior here.